src/features/singleRight/delegate/ChooseRightsPage/RightsActionBarContent/RightsActionBar.tsx

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import * as React from 'react';
+import { Paragraph, Heading, Chip, Alert } from '@digdir/design-system-react';
+import { useTranslation } from 'react-i18next';
+
+import type { IdValuePair } from '@/dataObjects/dtos/singleRights/DelegationInputDto';
+import { getSingleRightsErrorCodeTextKey } from '@/resources/utils/errorCodeUtils';
+
+import classes from './RightsActionBarContent.module.css';
+
+export type Right = {
+  action: string;
+  delegable: boolean;
+  checked: boolean;
+  resourceReference: IdValuePair[];
+};
+
+export interface RightsActionBarContentProps {
+  /** The callback function to be called when toggling a right. */
+  toggleRight: (serviceIdentifier: string, action: string) => void;
+
+  /** List of rights to be presented as toggle chips */
+  rights: Right[];
+
+  /** Description of the service */
+  serviceDescription?: string;
+
+  /** Description of the rights */
+  rightDescription?: string;
+
+  /** Unique identifier for the service the rights adhere to */
+  serviceIdentifier: string;
+
+  /** code representing the error if service contains rights that are not delegable */
+  errorCode?: string;
+}
+
+export const RightsActionBarContent = ({
+  toggleRight,
+  rights,
+  serviceDescription,
+  rightDescription,
+  serviceIdentifier,
+  errorCode,
+}: RightsActionBarContentProps) => {
+  const { t } = useTranslation('common');
+  const hasUndelegableRights = rights.some((r) => r.delegable === false);
+
+  const serviceResourceContent = (
+    <>
+      <Paragraph>{serviceDescription}</Paragraph>
+      <Paragraph>{rightDescription}</Paragraph>
+      <Paragraph>{t('single_rights.action_bar_adjust_rights_text')}</Paragraph>
+      <Heading
+        size={'xxsmall'}
+        level={5}
+      >
+        {t('single_rights.choose_rights_chip_text')}
+      </Heading>
+      <div className={classes.chipContainer}>
+        {rights
+          .filter((right) => right.delegable === true)
```

Comment:
Even though it might feel unneccessary here; state what the object is, even though the object is also Right. When stating it, it also validates that the object is in fact a Right